code/modules/mob/living/simple_animal/simple_animal_vr.dm

```diff
@@ -127,6 +129,8 @@
 /mob/living/simple_animal/death()
 	for(var/I in vore_organs)
 		var/datum/belly/B = vore_organs[I]
+		for(var/mob/living/M in B.internal_contents) // because release_all_contents doesn't release absorbed things
+			M.absorbed = 0
```

Comment:
I like all the changes except this one. I know you told me about this one and asked if that was the way to do it in Discord but sometimes it's hard to get context like that.

I think a better way to do it would be to add a parameter to release_all_contents, like
```
/datum/belly/proc/release_all_contents(var/include_absorbed = FALSE)
	if (internal_contents.len == 0)
		return 0
	for (var/atom/movable/M in internal_contents)
		if(!include_absorbed && istype(M,/mob/living))
			var/mob/living/ML = M
			if(ML.absorbed)
				continue
```
Then just call `B.release_all_contents(include_absorbed = TRUE)`. It will be faster.

Reply:
Fixed as per discord convo to avoid problems of people being released with absorbed still true and stuck with weldervision forever